engine/src/main/java/org/hibernate/search/backend/impl/DeletionQueryMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * @author Martin Braun
+ */
+public interface DeletionQueryMapper {
+
+	Query toLuceneQuery(DeletionQuery deletionQuery, ScopedAnalyzer analyzerForEntity);
```

Comment:
Hmm. I kinda understand what you mean now.

At the moment serialization is done by storing the key of a query that can 100% be matched back to a class. But this class doesn't have to stay the same between Hibernate-Search versions, enabling us to change the implementations between versions however we'd like.

To get this straight, you want a (atm straight forward) switch-case block (to remove the need for a array lookup) combined with manual calls to the right fromString() method. I guess this logic can then stay in DeleteByQuerySupport until Hibernate-Search uses Java8 (then we can move it to the DeletionQuery interface).